internal/controller/spireserver_controller_mock_test.go

```diff
@@ -107,6 +107,16 @@ func TestSpireserverController(t *testing.T) {
 	}
 }
 
+func TestCheckTrustDomain(t *testing.T) {
+	validTrustDomain := checkTrustDomain("trustdomain")
```

Comment:
While `trustdomain` will not return an error with `checkTrustDomain` since it is valid given how `spiffeid.TrustDomainFromString` is written, I would suggest adding test cases that check whether valid DNS names are passing too. This is because trust domains take the same form as DNS names (ref: https://spiffe.io/docs/latest/deploying/configuring/#configuring-a-trust-domain).

Reply:
Thank you for pointing this out. I will look into it!